source/frontend/src/features/mapSideBar/disposition/form/DispositionForm.tsx

```diff
@@ -42,6 +51,29 @@ const DispositionForm = React.forwardRef<FormikProps<DispositionFormModel>, IDis
       API.DISPOSITION_PHYSICAL_STATUS_TYPES,
     );
 
+    const onMinistryProjectSelected = React.useCallback(
+      async (param: IAutocompletePrediction[]) => {
+        if (param.length > 0) {
+          if (param[0].id) {
+            const result = await retrieveProjectProducts(param[0].id);
+            if (result) {
+              setProjectProducts(result as unknown as ApiGen_Concepts_Product[]);
```

Comment:
I still don't understand why all this casting is necessary. Is this not the type returned by the API?

Reply:
same as previous comment. Related to the use of the new generated models but limiting to the scope of what I am working on.